### DISCUSSION TITLE
[Proposal] C# Feature Support partial enum or enum Inherit

### DISCUSSION THREAD

**Discussion by wangronghua** [dotnet/csharplang#7969] [Language Ideas]
Now there are three modules each responsible for three people, with a log type enumeration. Each module will have multiple tables, and each table corresponds to an enumeration item in the log type enumeration. A partial for the enumeration type is needed, so there is no need for multiple people to modify the same enumeration file
```
        public partial enum LogType : byte
        {
            ...
        }
```
or

```
        public enum LogType : BaseLogType
        {
            ...
        }
```

Inherit is based on the following requirements. In the basic class library, an error type enumeration will be used, which defines various error types. This enumeration will also be extended in the business module to include some business-related error types

**Comment by colejohnson66** [comment]
Enumerations have never supported additional values by consumers, and probably never will. It's too complicated a thing for such niche cases. If you want a way to "register" things in your library, a GUID to whatever dictionary (`Dictionary<Guid, MyCallback>` or whatever) would be your best choice. Also, `partial enum` has been proposed before: #2669.
